javaagent-bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/AgentInitializer.java

```diff
@@ -34,21 +34,27 @@ private static synchronized void startAgent(Instrumentation inst, URL bootstrapU
       throws Exception {
     if (AGENT_CLASSLOADER == null) {
       ClassLoader agentClassLoader = createAgentClassLoader("inst", bootstrapUrl);
+      AGENT_CLASSLOADER = agentClassLoader;
+
       Class<?> agentInstallerClass =
           agentClassLoader.loadClass("io.opentelemetry.javaagent.tooling.AgentInstaller");
```

Comment:
If `agentClassLoader` is an `ExtensionClassLoader` then won't we put `AgentInstaller` in the same CL as extensions? And won't it apply shading transformations on all agent classes?

Reply:
No, because `ExtensionClassLoader` first delegates to parent, as usual. And its parent is the old `AgentClassLoader`